spec/unit/jobs/v3/services/update_broker_job_spec.rb

```diff
@@ -182,8 +182,8 @@ module V3
           describe 'when there is a model level validation error' do
             before do
               allow_any_instance_of(ServiceBroker).to receive(:validate) do |record|
-                # Make setting the state to 'AVAILABLE' fail, but nothing else
-                if record.state == 'AVAILABLE'
+                # Make setting the name to 'new-name' fail, but nothing else
+                if record.name == 'new-name'
```

Comment:
This test unfortunately tested the only case in which the state of the SB could be reset to its previous state after a failed model validation. This is because in the rescue call, the broker's state will be explicitly set to the previous state and will overwrite the state `AVAILABLE` which caused the model violation to fail.
If the cause for the model violation is something else, e.g. a name, then the test would fail, because the broker object would still have the name `new-name` which caused the model violation. This leads to a failed broker.update which is meant to reset the brokers state to its previous state.